### DISCUSSION TITLE
Optimizer on chimney height (wall thickness)?

### DISCUSSION THREAD

**Discussion by Sonoton** [edwardkort/WWIDesigner#158] [General]
Can an optimization be conducted on chimney height (wall thickness of toneholes).  I am thinking of  cases where a fixed bore diameter is used, with a given fixed set of tonehole diameters, and hole spacings limited within certain ranges.   A wall thickness optimizer at the tonehole locations might allow a more optimum design in some cases.   Is this already built into WWIDesigner, or does this have to be achieved by rerunning the program for different sets of wall thickness variations.  Thank you.

**Comment by bhp1** [comment]
WIDesigner does not currently have an optimizer for tonehole height / wall thickness.  If you're looking at a uniform wall thickness, I'm not sure you'd really see an optimum beyond either "as thin as possible" or "as thick as possible".  If this is the case, you can get the information you need by running an optimization with the thinnest and thickest feasible walls, and perhaps one value in between.

**Comment by edwardkort** [comment]
Benade postulated that a woodwind with uniform tonehole sizes would produce notes with a more consistent timbre. He suggested to accomplish this by varying the hole height (wall thickness in the case of a NASF). Since a uniform timbre is one of my goals when making NASFs, I ran a test - manually changing the wall thickness at each hole. I then built two flutes: one with uniform hole size with varying wall thicknesses, the other with a uniform wall thickness and varying tonehole diameters. I constructed these flutes using 3D printing, guarantying a couple of thousandths of an inch accuracy.

The results were less than striking. I could detect no timbre differences between the two flutes. This greatly reduced my desire to program an optimizer strategy to automate this idea.

**Comment by Sonoton** [comment]
Thank you for your very considered and prompt replies.

Not actually being a flautist and being a novice in music, I really appreciate your insight. I believe I have (only) some slight appreciation of timbre--my question stemmed from observing in the Hoejke and Goss spreadsheets, that changing chimney height allows additional movement of the toneholes, in order to accommodate smaller interdigital spacing, while allowing more playability (as in matching tuning via breath curves).

Being a novice, please let me ask three additional questions. 
1) Does WWiDesigner somehow have breath "curves", e.g. breath curve zero, breath curve one, etc internally in the coding--in using a breath curve how does one know what is using during actual playing? 
2) Is it possible to stipulate a constraint that the "green dots" match a given splay of "open blue circles" in the tuning graph as a new feature for WWiDesigner to optimize as playability. 
3) If a flautist wanted to overcome a deficiency in timbre, would they compensate somehow, as "cover it up" in a Boehm-flute by playing with more "breathy-ness", or could a NAF flautist try to play pianissimo and use studio amplification? I am assuming nonuniformity in timbre has the possible implication that some notes would sound more ragged than others.

Thank you for your time. 
Best Regards!
     On Monday, March 25, 2024 at 02:48:42 PM MDT, Edward Kort ***@***.***> wrote:  
 
 


Benade postulated that a woodwind with uniform tonehole sizes would produce notes with a more consistent timbre. He suggested to accomplish this by varying the hole height (wall thickness in the case of a NASF). Since a uniform timbre is one of my goals when making NASFs, I ran a test - manually changing the wall thickness at each hole. I then built two flutes: one with uniform hole size with varying wall thicknesses, the other with a uniform wall thickness and varying tonehole diameters. I constructed these flutes using 3D printing, guarantying a couple of thousandths of an inch accuracy.

The results were less than striking. I could detect no timbre differences between the two flutes. This greatly reduced my desire to program an optimizer strategy to automate this idea.

**Comment by bhp1** [comment]
I can't comment on timbre, but for the whistle and flute study models WIDesigner does have an option to control the breath curve (the pattern of open blue circles).  This option is called the [blowing level](https://github.com/edwardkort/WWIDesigner/wiki/WIDesigner-Options#blowing-level).  When you perform an optimization, WIDesigner is trying to adjust the geometry to get the predicted nominal playing frequencies (the open blue circles) close to the target playing frequencies (shown with green, yellow or red markers).
